hbase-server/src/main/java/org/apache/hadoop/hbase/master/janitor/CatalogJanitor.java

```diff
@@ -165,13 +168,19 @@ public int scan() throws IOException {
       this.lastReport = scanForReport();
       if (!this.lastReport.isEmpty()) {
         LOG.warn(this.lastReport.toString());
+      } else {
+        LOG.debug(this.lastReport.toString());
```

Comment:
nit: toString() might be an expensive call if debug is not enabled, might want to switch to a isDebugEnabled() guard, (later versions of log4j supports log.debug(Object)/log.debug(Supplier) so that toString() is eventually evaluated after the debug guard check)..

Reply:
+1

Reply:
We used to abide by that guidance but I don't see that done often in new code.